sycl/doc/extensions/DeviceGlobal/SYCL_INTEL_device_global.asciidoc

```diff
@@ -0,0 +1,790 @@
+
+Returns a reference to the underlying `T` on the device. It is undefined behavior to access the reference or any address derived from it on a different device or on the host.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+device_global& operator=(const T&) noexcept;
+----
+|
+Available only in device functions.
+
+Enables assignment of type `T` to the underlying allocation on the device.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+element_type& operator[]( std::ptrdiff_t idx ) noexcept;
+const element_type& operator[]( std::ptrdiff_t idx ) const noexcept;
+----
+|
+Available only in device functions.
+
+Available only when the underlying `T` defines an `operator[]`.
+
+Indexes into the underlying `T`. It is undefined behavior if _idx_ is negative.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+T& operator->() noexcept;
+const T& operator->() const noexcept;
+----
+|
+Available only in device functions.
+
+Available only when `+operator->+` is valid for objects of type `T`.
+
+Provides member access through `T` that is a pointer or a class which defines `+operator->+`.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+template<typename propertyT>
+static constexpr bool has_property();
+----
+| Returns true if the `property_listT` contains the property specified by `propertyT`. Returns false if it does not.
+Available only if `sycl::is_property_of_v<propertyT, sycl::ext::oneapi::device_global>` is true.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+template<typename propertyT>
+static constexpr auto get_property();
+----
+| Returns an object of the class used to represent the value of property `propertyT`.
+Must produce a compiler diagnostic if `property_listT` does not contain a `propertyT` property.
+Available only if `sycl::is_property_of_v<propertyT, sycl::ext::oneapi::device_global>` is true.
+
+|===
+
+=== Relax language restrictions for SYCL device functions
+
+SYCL 2020 restrictions must be relaxed to allow `device_global` to be used within
+device functions without being `const` or `constexpr` and without being zero-initialized
+or constant-initialized.  This is achieved by adding `device_global` exceptions to the
+following point in Section 5.4 "Language restrictions for device functions".  The modified restriction is:
+
+* Variables with static storage duration that are odr-used inside a device function, must be
+`const` or `constexpr` and zero-initialized or constant-initialized, except if the variable is
+of type `device_global` in which case it can be odr-used inside a device function without being
+`const`/`constexpr` or zero-/constant-initialized.
+** Amongst other things, this restriction makes it illegal for a device function to access a
+global variable that isn’t `const` or `constexpr` unless the variable is of type `device_global`.
+
+
+=== Add new copy and memcpy members to the queue class
+
+Add the following functions to the `sycl::queue` interface described in Section 4.6.5.1 of
+the SYCL 2020 specification.
```

Comment:
Reading through the table beneath I was still surprised by the presence of the `startIndex` and `offset` arguments.  Maybe a non-normative `NOTE:` here that reminds readers that there's no way to perform pointer arithmetic directly on the `device_global` would be useful here?

Reply:
Agreed.  Added a note.